tools/compatibility-check/contracts_checker.go

```diff
@@ -117,7 +117,10 @@ func (c *ContractsChecker) analyze(
 	config *analysis.Config,
 	locations []common.Location,
 ) {
-	programs := make(analysis.Programs, len(locations))
+	programs := analysis.Programs{
+		Programs:                  make(map[common.Location]*analysis.Program, len(locations)),
+		CryptoContractElaboration: config.CryptoContractElaboration,
```

Comment:
This currently doesn't work, since the tool doesn't have the access to Crypto contract's elaboration. Not sure what the correct solution is though. Maybe we would have to get it from FVM (adds a dependency to fvm).

Reply:
Yeah, we'll likely need to pull in FVM, which is not ideal, but needed if we want to check contracts exactly like they are checked on-chain

Reply:
@SupunS I can help with / take this over if you want to

Reply:
I pushed this fix https://github.com/onflow/cadence/pull/3631/commits/cfd6412917aee4a9d78265a8eccc935b7bb2daba, to only convert the `Crypto` import to a proper address-qualified import, so that it will resolve Crypto as any other contract. But unfortunately that requires the Crypto contract to be deployed on chain. So this will still be broken until the next network upgrades.

Reply:
@turbolent if you know an easy way to get the elaboration from the FVM directly, that would be great!

Reply:
For now, I provided the `Crypto` contract's code manually for the tool (https://github.com/onflow/cadence/pull/3631/commits/871780ed35abc62eb3fe5795d68ee39e6e4d23fa). Left a TODO: to revert it once the contract is deployed on-chain.